### PR TITLE
CCM 11966 : Add migration to set saving for older governance rule executions

### PR DESCRIPTION
**Pull request by sandyydk**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46559)

### CODE REVIEW

File: 450-ce-views/src/main/java/io/harness/ccm/views/entities/RuleExecution.java [hunk 1]
```diff
@@ -119,6 +121,8 @@ public RuleExecution toDTO() {
         .lastUpdatedAt(getLastUpdatedAt())
         .executionStatus(getExecutionStatus())
         .ruleEnforcementName(getRuleEnforcementName())
+            .realizedSavings(getRealizedSavings())
```
Comment (sahibaCCM) [inline 450-ce-views/src/main/java/io/harness/ccm/views/entities/RuleExecution.java:124]: I think you don't have clang format installed

File: ce-nextgen/service/src/main/java/io/harness/ccm/migration/GovernanceSavingsMigration.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @Inject private HPersistence hPersistence;
+    @Inject private RuleExecutionDAO ceRuleExecutionRecordDao;
+
+    @Override
+    public void migrate() {
+
+        try {
+            log.info("Starting migration (updates) of all CE Governance Rule Executions");
+            final List<RuleExecution> ruleExecutionRecords =
+                    hPersistence.createQuery(RuleExecution.class, excludeAuthority).asList();
```
Comment (sahibaCCM) [inline ce-nextgen/service/src/main/java/io/harness/ccm/migration/GovernanceSavingsMigration.java:36]: Can we move HPersistence  query to RuleExecutionDAO?

File: 450-ce-views/src/main/java/io/harness/ccm/views/entities/RuleExecution.java (hunk 1 above)
Comment (sandyydk) [reply 450-ce-views/src/main/java/io/harness/ccm/views/entities/RuleExecution.java]: This is fixed

**Comment by nikunj-badjatya** [comment]
Lets park this for now.

**Pull request closed without merging**